cdap-ui/app/cdap/components/AbstractWidget/SchemaEditor/Context/SchemaGenerator.ts

```diff
@@ -0,0 +1,240 @@
+  }
+  if (typeProperties.aliases) {
+    finalType.aliases = typeProperties.aliases;
+  }
+  if (Array.isArray(children.order)) {
+    for (const childId of children.order) {
+      const currentChild = children[childId];
+      const { name, type, nullable: isFiledNullable } = currentChild;
+      if (!name || name === '') {
+        continue;
+      }
+      const isFieldTypeComplex = isDisplayTypeComplex({ type });
+      if (!isFieldTypeComplex) {
+        finalType.fields.push({
+          name,
+          type: isFiledNullable ? [type, 'null'] : type,
+        });
+      } else {
+        finalType.fields.push({
+          name,
+          type: generateSchemaFromComplexType(
+            currentChild.type,
+            currentChild,
+            currentChild.nullable
+          ),
+        });
+      }
+    }
+  }
+  return nullable ? [finalType, 'null'] : finalType;
+}
+
+function generateUnionType(children: IOrderedChildren) {
+  const finalType = [];
+  if (Array.isArray(children.order)) {
+    for (const childId of children.order) {
+      const currentChild = children[childId];
+      const { type } = currentChild;
+      const isUnionTypeComplex = isDisplayTypeComplex({ type });
+      if (!isUnionTypeComplex) {
+        finalType.push(type);
+        continue;
+      }
+      finalType.push(generateSchemaFromComplexType(type, currentChild, false));
+    }
+  }
+  return finalType;
+}
+
+function generateLogicalType(child) {
+  const { typeProperties, nullable } = child;
+  return nullable ? [typeProperties, 'null'] : typeProperties;
+}
+
+function generateSchemaFromComplexType(type: string, currentChild, nullable: boolean) {
+  const complexTypeChildren: IOrderedChildren = currentChild.children;
+  switch (type) {
+    case 'array':
+      return generateArrayType(complexTypeChildren, nullable);
+    case 'map':
+      return generateMapType(complexTypeChildren, nullable);
+    case 'enum':
+      return generateEnumType(complexTypeChildren, currentChild, nullable);
+    case 'union':
+      return generateUnionType(complexTypeChildren);
+    case 'record':
+      return generateRecordType(complexTypeChildren, currentChild, nullable);
+    case 'time':
+    case 'timestamp':
+    case 'decimal':
+    case 'date':
+      return generateLogicalType(currentChild);
+    default:
+      return type;
+  }
+}
+
+/**
+ * Utility to convert the entire schema tree to a valid avro schema JSON.
+ * @param schemaTree Schema tree to convert to avro schema JSON.
+ */
+function SchemaGenerator(schemaTree: INode) {
+  const avroSchema: ISchemaType = getDefaultEmptyAvroSchema();
+  if (!schemaTree) {
+    return avroSchema;
+  }
+  // Top level record fields.
+  const { order } = schemaTree.children;
+  if (Array.isArray(order)) {
+    for (const id of order) {
+      const currentField = schemaTree.children[id];
+      const { name, type, nullable } = currentField;
+      // Skip the newly added rows.
+      if (!name || name === '') {
```

Comment:
I believe if (!name) will suffice since this will check for empty string as well.